src/module.c

```diff
@@ -4235,7 +4240,9 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
          * recursive call to this module.) */
         c = createClient(NULL);
     }
-    c->user = NULL; /* Root user. */
+
+    /* Set a root user or the client module user if the module asked to check ACL permissions */
+    c->user = (flags & REDISMODULE_ARGV_CHECK_ACL) ? ctx->client->user : NULL; /* Root user. */
```

Comment:
i'm not certain what are the side effects of this.
maybe instead we should keep `c->user` set to NULL, and use a more explicit code below to call `ACLCheckAllUserCommandPerm` instead of `ACLCheckAllPerm`

@MeirShpilraien @guybe7 can you think of any undesired or desired side effects?

Reply:
I agree, even if there are no desired side effects now I can see how they pop up in the future.

Reply:
Another point is what we do if the context has no associated user. If we want to be more secure by default, I'd say we may need to fail the operation.